lib/routes/routes.dart

```diff
@@ -520,30 +520,6 @@ class HomeRouter extends RouterDelegate<HomeRoutes<Object?>>
       !(_currentSearchDelegate?.chipsBarDragged ?? false);
   }
 
-  /// Callback that must be called before any pop.
-  /// 
-  /// For example we want that player route would be closed first.
-  bool handleNecessaryPop() {
```

Comment:
Refactoring it like you did breaks pop behavior for both of them on any route, except the starting one - search route, playlist route, artist route and there might be more.

You just had to move the whole `handleNecessaryPop` to the same place you moved the part of it - `TabsRouteState ._handlePop`, instead of splitting it into individual BackButtonListeners.

This overall is not needed for to fix the issue, but it is an OK refactor.